sycl/doc/extensions/Assert/level-zero.md

```diff
@@ -0,0 +1,19 @@
+# Overview
+
+This extension enables detection of assert failure of kernel.
+
+# New enum value
+
+`ze_result_t` enumeration should be augmented with `ZE_RESULT_ABORTED` enum
+element. This enum value indicated a detected assert failure at device-side.
+
+# Changed API
+
+```
+ze_event_handle_t Event; // describes an event of kernel been submitted previously
+ze_result Result = zeEventQueryStatus(Event);
+```
+
+If kernel failed an assertion `zeEventQueryStatus` should return
```

Comment:
I don;t think this is possible to achieve in asynchronous / non-blocking way in L0.

We dont have any communication between kernel and event - so we can;t signal events with "assert happened" information.

if we use global / program wide assert buffer - each kernel will be using the same assert happened flag  - we do not have fine grain control to determine which kernel - and which connected event fired the assert.


Fences could be used - allowing to synchronize at cmdQueue level and not kernel - any kernel causing assert executed in cmd Queue can then make fence synchronize to return error:https://spec.oneapi.com/level-zero/latest/core/PROG.html#fences

Reply:
Is it still possible in OpenCL.
Can the OpenCL approach be reused in Level-Zero?

Reply:
Could you, please, provide more details about using fences?

Reply:
fences are decribed in L0 spec - they are similar to events, but directly connected to command queues: https://spec.oneapi.com/level-zero/latest/core/PROG.html#fences

In OpenCL the submission model is different - each enqueue is independent - single kernel is submitted ( queued)  at a time. L0 operates on command lists that may contain multiple kernels - once cmd list is submitted to HW - we can;t control when a kernel in whole sequence is started completed.

OpenCL handles kernels with printf in a blocking way - enqueueNDRangeKErnel with printf makes this a blocking call - so we have fine control when specific kernel is completed  - we can do the same for assert() message - output event will be created when the kernel has already finished. I L0 this is not possible - as we would have to synchronize whoel command list.